tests/jobs/test_local_task_job.py

```diff
@@ -799,6 +800,121 @@ def test_mini_scheduler_works_with_wait_for_upstream(self, caplog, dag_maker):
         assert failed_deps[0].dep_name == "Previous Dagrun State"
         assert not failed_deps[0].passed
 
+    @pytest.mark.parametrize(
+        "exception, trigger_rule",
+        [
+            (AirflowFailException(), TriggerRule.ALL_DONE),
+            (AirflowFailException(), TriggerRule.ALL_FAILED),
+            (AirflowSkipException(), TriggerRule.ALL_DONE),
+            (AirflowSkipException(), TriggerRule.ALL_SKIPPED),
+            (AirflowSkipException(), TriggerRule.NONE_FAILED_MIN_ONE_SUCCESS),
+        ],
+    )
+    @conf_vars({('scheduler', 'schedule_after_task_execution'): 'True'})
+    def test_mini_scheduler_works_with_skipped_and_failed(self, exception, trigger_rule, caplog, dag_maker):
+        """
+        In these cases D is running, at no decision can be made about C.
+        """
+
+        def raise_():
+            raise exception
+
+        session = settings.Session()
```

Comment:
You can use the `session` fixture here instead, which automatically calls `rollback` on test teardown.